core-bundle/src/Migration/AbstractRecordedMigration.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Migration;
+
+use Contao\CoreBundle\Entity\Migration as MigrationEntity;
+use Doctrine\DBAL\Connection;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Tools\SchemaTool;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
+use Symfony\Contracts\Service\ServiceSubscriberTrait;
+
+abstract class AbstractRecordedMigration extends AbstractMigration implements ServiceSubscriberInterface
+{
+    use ServiceSubscriberTrait;
+
+    public function shouldRun(): bool
+    {
+        return !$this->hasRun();
+    }
+
+    protected function hasRun(): ?bool
+    {
+        if (!$this->connection()->getSchemaManager()->tablesExist(['tl_migration'])) {
+            return false;
+        }
+
+        return null !== $this->entityManager()->getRepository(MigrationEntity::class)->findOneBy(['name' => $this->getName()]);
+    }
+
+    protected function createResult(bool $successful, string $message = null): MigrationResult
+    {
+        if (!$this->hasRun()) {
```

Comment:
Should we always create an entity if the migration was run, even if it runs multiple times?

Reply:
I think we should, yes. If your `shouldRun` relies on this information, then it must be set whenever it is run once. Though may be `createResult` is the wrong location for this to happen and it should rather be its own dedicated method, that you have to call yourself separately?

Reply:
`createResult()` is OK for me to store the run information.